### PR TITLE
Adding attribution to places where an OpenLayers.Map gets created

### LINKED ISSUE #916: missing attribution “© OpenStreetMap contributors”. on the (osm?) map?
Is there a missing  attribution  _“© OpenStreetMap contributors”_ information on the (osm?) map for webpage: https://openflights.org/?

For additional information please see here: https://www.openstreetmap.org/copyright/en

### PR DESCRIPTION
**Pull request by kmpoppe**
Closes #916 .

Hey Jani,

I thought I could just give you a PR for the attribution, no need to make you go the long way round.

Please TRY IT before merging (check out my branch) because due to time issues I couldn't set up everything to try it myself.

HTH.

Kai

### CODE REVIEW

**Comment by reedy** [comment]
For the lazy, how would I test this? Or rather, how/where would the copyright attribution for the map actually appear? :)

**Comment by kmpoppe** [comment]
Hi there,

you will see the attribution in the lower right-hand corner of the map with the text (© openstreetmap contributors).

I should work worry-free, if you have a local version set up you should easily be able to spot it before publishing it :)

K

**Comment by reedy** [comment]
Screenshot 2023-03-01 at 15 30 42

With your patch cherry-picked onto master, I don't see anything extra on the map in terms of attribution...

File: openflights.js [hunk 1]
```diff
@@ -106,6 +106,10 @@ function init(){
 
   map = new OpenLayers.Map('map', {
     center: new OpenLayers.LonLat(0, 1682837.6144925),
+	attribution: {
+		title: "&copy; OpenStreetMap contributors",
+		href: "https://www.openstreetmap.org/"
+	},
```
Comment (reedy) [inline openflights.js:112]: I think, looking at line 134 below, this needs to be under a layer...

See also like https://github.com/openlayers/openlayers/compare/release-2.11...v2.13.1#diff-99e6b60d1f8e64080689545ad1091989875914a0d2b7bde885525413682473beR13-R17

And maybe an attribution control adding?

**Review by reedy — CHANGES_REQUESTED** [review]
Doesn't seem to work in my testing, and while the json blob for the attribution looks right, it doesn't look like it's in the right place or under the right object

File: openflights.js (hunk 1 above)
Comment (reedy) [reply openflights.js:112]: I'm also curious which layer(s) the OSM attribution needs to be against too. Especially as below the "Artistic", "Political" and "Satellite" have attribution...

But I also cannot see where those attributions appear on the maps either.

**Comment by reedy** [comment]
Finding appropriate documentation is made harder by OpenFlights using a version of OpenLayer from ~November 2011. Seemingly version 2.11. 7.3 is now out.

Ping https://github.com/jpatokal/openflights/issues/1192

**Comment by reedy** [comment]
https://github.com/openlayers/openlayers/blob/release-2.11/examples/attribution.html would be the relevant docs...

File: openflights.js (hunk 1 above)
Comment (reedy) [reply openflights.js:112]: Ok, and the key is for those attributions not appearing...

https://github.com/openlayers/openlayers/blob/release-2.11/examples/attribution.html#L31-L34

Screenshot 2023-04-23 at 16 40 00

Screenshot 2023-04-23 at 16 45 22

It's not pretty... But it's a start...

```
diff --git a/openflights.js b/openflights.js
index 8c2db2d..9e43d17 100644
--- a/openflights.js
+++ b/openflights.js
@@ -115,7 +115,8 @@ function init(){
       new OpenLayers.Control.Navigation({'title': gt.gettext("Toggle pan and region select mode")}),
       new OpenLayers.Control.LayerSwitcher({'ascending':false, 'title': gt.gettext('Switch map layers')}),
       new OpenLayers.Control.ScaleLine(),
-      new OpenLayers.Control.OverviewMap({'title': gt.gettext("Toggle overview map")})
+      new OpenLayers.Control.OverviewMap({'title': gt.gettext("Toggle overview map")}),
+      new OpenLayers.Control.Attribution()
     ] });
 
   // Horrible hack to stop OpenLayers 2 from showing ZL < 2
```

Need to play around with the location of things...

Comment (reedy) [reply openflights.js:112]: Which looks to be done in CSS - https://stackoverflow.com/questions/20981679/openlayers-map-change-position-of-attribution-in-javascript

```
.olControlAttribution {
    font-size: smaller; 
    right: 3px; 
    bottom: 4.5em; 
    position: absolute; 
    display: block;
}
```

Comment (reedy) [reply openflights.js:112]: Hmm. And after https://github.com/jpatokal/openflights/commit/a5c352206c7d31fe59ad2349a34c3b4597dd5a15 it seems to be working as expected...

Yeah, the colours suck on some of the maps...

Screenshot 2023-04-23 at 17 03 33
Screenshot 2023-04-23 at 17 03 36
Screenshot 2023-04-23 at 17 03 39

**Comment by reedy** [comment]
May no longer be needed/superseded by https://github.com/jpatokal/openflights/commit/b82c1443501a7f80eabb0cf356d5402c8b114a39

**Pull request closed without merging**